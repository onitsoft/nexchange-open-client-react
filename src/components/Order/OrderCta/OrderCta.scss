.container.container {
  background: $green;

  @media (min-width: $screen-md-min) {
    padding: 30px 65px;
  }
}

.title {
  margin: 0;
  font-size: 20px;
  font-weight: 400;
  color: #fff;

  @media (min-width: $screen-md-min) {
    margin-top: 10px;
    font-size: 22px;
    font-weight: 500;
  }

  @media (min-width: $screen-lg-min) {
    margin-top: 15px;
  }

  span {
    text-transform: capitalize;
  }
}

.form-group {
  position: relative;

  @media (min-width: $screen-sm-min) {
    &:hover {
      .btn.btn.btn {
        border: 1px solid $green;
        background: $green;
        color: #fff;
        box-shadow: $btn-box-shadow;
      }
    }
  }

  input {
    width: 100%;
    height: 60px;
    font-size: 16px;
    margin-top: 15px;
    border-radius: 4px;
    padding: 0 15px;

    @media (min-width: $screen-md-min) {
      margin-top: 20px;
      margin-bottom: 15px;
      padding: 0 30px;
    }
  }
}

.referral-input.referral-input {
  color: $green;
  cursor: pointer;
}

.btn.btn.btn {
  margin: 15px 0 5px 0;
  width: 100%;
  border: 1px solid #fff;

  &:hover {
    border: 1px solid #fff;
  }

  @media (min-width: $screen-ms-min) {
    position: absolute;
    top: 30px;
    right: 14px;
    width: auto;
    margin: 0;
    padding: 6px 15px;
    font-size: 12px;
    
    color: $green;
    border: 1px solid $green;
    background-color: #fff;

    &:hover {
      border: 1px solid $green;
    }
  }

  @media (min-width: $screen-md-min) {
    right: 30px;
    top: 35px;
  }
}

.text.text {
  @media (min-width: $screen-md-min) {
    padding-right: 60px;
  }
}

.img {
  max-width: 100px;
  margin: 5px 0 20px 0;

  @media (min-width: $screen-ms-min) {
    max-width: 150px;
  }

  @media (min-width: $screen-sm-min) {
    max-width: 180px;
    margin: 5px 0;
  }

  @media (min-width: $screen-md-min) {
    max-width: 210px;
  }
}

.message {
  color: #fff;
}

.share {
  margin-top: 24px;
  color: #fff;
  font-size: 18px;
}

.links {
  margin-top: 10px;

  @media (min-width: $screen-sm-min) {
    display: inline-block;
    margin-top: 0;
    margin-left: 15px;
  }
}

.link {
  display: inline-block;
  position: relative;
  width: 40px;
  height: 40px;
  margin: 0 5px;
  border: 1px solid #fff;
  border-radius: 40px;
  text-align: center;

  i {
    margin-top: 9px;
    color: #fff;
  }
}