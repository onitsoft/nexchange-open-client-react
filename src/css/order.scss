@import "variables";
@import "mixins";
@import "bootstrap/variables";
@import "bootstrap/mixins";


#order { background: #f8f7f8; padding: 15px 0 0 0;
	a { color: $green; 
		&:visited, &:focus { text-decoration: none; }
		&:hover { text-decoration: underline; }
	}
	#order-ref { margin-bottom: 15px; }
	.box {
		@media (min-width: $screen-sm-min) { margin-bottom: 30px; }
		
		h4 { font-size: 15px; }
		h5, h6 { margin: 0; text-transform: none; }
		.coin-icon { position: relative; top: 8px; font-size: 28px; }
		.text-warning { color: $green; }		
	}
	.coin-box { padding-left: 10px; }
	#time-remaining { color: #2cb4a0; }

	@media (min-width: $screen-sm-min) { padding: 25px 0;
		h5 { font-size: 15px; }
		h6 { font-size: 12px; }
		.box {
			h4 { font-size: 18px; }
			.coin-icon { top: 12px; }
		}
	}

	@media (min-width: $screen-md-min) { padding: 50px 0 0 0;
		h5 { font-size: 17px; }
		h6 { font-size: 16px; }
		.box .coin-icon { top: 12px; }
	}
}

#order-payment { text-align: center; padding-top: 10px; overflow: hidden;
	h3, h4 { font-size: 20px;}
	#deposit-address { word-wrap: break-word; }
	img { margin-top: -15px; max-width: 150px;
		@media (min-width: $screen-sm-min) { margin-top: -7px; }
	}
	#order-payment-details { margin-top: 10px;
		h3 { margin-top: 0; }
		@media (min-width: $screen-sm-min) { margin-left: -15px; }
		@media (min-width: $screen-md-min) { margin-top: 10px; }
		@media (min-width: $screen-lg-min) { margin-top: 30px; }
	}
	.box { padding-bottom: 5px; }

	@media (min-width: $screen-sm-min) { text-align: left; padding-left: 13px;
		img { max-width: 100%; }
		h3 { font-size: 24px; }
	}
	@media (min-width: $screen-md-min) { padding-left: 50px; }
	@media (min-width: $screen-lg-min) { padding-left: 60px; }
}

#order-status { padding: 8px 0 5px 0; overflow: hidden;
	hr { margin-top: 5px; }
	.step { width: 100%; margin-top: 15px; text-align: center;
		h4 { margin: 0; background: #fff; text-transform: uppercase; font-size: 15px; font-weight: 500; color: #adadad; }
		.glyphicon { font-size: 30px; color: #adadad; }
		&.active h4, &.active .glyphicon {
			@include animation(color-change 1s infinite alternate);
		}
		&.done h4, &.done .glyphicon { color: $green; }

		@media (min-width: $screen-sm-min) { width: 33.33%; float: left; }
	}
	.progres-container { width: 100%; padding-top: 25px; clear: both;
		.progress .progress-bar.progress-bar-info { background: $green; }
		.progress.progress-line-info { background: rgba(44, 180, 160, 0.2); }
		.progress.progress-line-info { margin-bottom: 5px; }

		@media (min-width: $screen-md-min) { padding: 25px 0 0; }
	}
}

#order-header {
	#bookmark-button { font-size: 20px; padding: 0; margin-top: 0; color: $green; }

	@media (min-width: $screen-sm-min) {
		h3 { float: left; }
		#bookmark-button { float: right; margin-top: 22px; }
	}
}

#bookmark {
	.bookmark-text{ font-size: 19px; }
	.input-group-addon { cursor: pointer; padding-top: 0;
		i { font-size: 18px; }
	}
	.form-group { margin-top: 10px; }
	input { font-size: 18px; }
}

.order-status-section {
	h2 { font-size: 24px; }

	@media (min-width: $screen-sm-min) {
		h2 { font-size: 30px; }
	}
}

#share-referral, #notifications { text-align: center;
	.box { padding: 20px; }
	.share { margin-top: 15px; }
	.fa { font-size: 45px; margin: 0 15px 0 0; color: #d6d6d6; }
	h2 { margin-top: 0; font-size: 22px;
		@media (min-width: $screen-sm-min) { font-size: 26px; }
		@media (min-width: $screen-md-min) { font-size: 34px; }
	}
}

#notifications {
	.form-group { margin-top: 10px; }
	button { width: 100%; margin-bottom: 0; }
	@media (min-width: $screen-sm-min) { button { width: auto; } }
}

@include keyframes(color-change) {
	from { color: #adadad; }
	to { color: #fbc02d; } 
}
