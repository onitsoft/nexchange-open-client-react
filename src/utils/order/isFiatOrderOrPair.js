const isFiatOrderOrPair = orderOrPair =>
    orderOrPair?.pair ? !orderOrPair?.pair.quote?.is_crypto : !orderOrPair?.quote?.is_crypto

export default isFiatOrderOrPair
