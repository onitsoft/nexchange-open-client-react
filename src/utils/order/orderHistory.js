import { parseCookies } from 'nookies'

const mapToOrderHistoryDto = (createdOrderData, username) => {
    const { referralToken = '' } = parseCookies()

    const {
        token: anonymousAuthToken,
        unique_reference: orderId,
        amount_quote,
        amount_base,
        pair,
        created_on,
        referral_code,
    } = createdOrderData

    const mappedOrderDto = {
        userData: {
            username,
        },
        id: orderId,
        quote: pair.quote.code,
        base: pair.base.code,
        amount_quote: parseFloat(amount_quote),
        amount_base: parseFloat(amount_base),
        withdraw_address: '',
        created_on,
        statusName: '',
        userRefCode: referral_code?.[0]?.code,
        referralToken,
    }

    return { mappedOrderDto, anonymousAuthToken, orderId }
}

const saveOrderHistory = orderHistory => {
    localStorage.setItem('orderHistory', JSON.stringify(orderHistory))
}

const saveCreatedOrderInStorage = mappedOrderDto => {
    // Comment: look for order history array on localStorage (if not present create an empty array)
    // and add the order details to top of array
    const orderHistory = localStorage.orderHistory ? JSON.parse(localStorage.orderHistory) : []

    orderHistory.push(mappedOrderDto)
    saveOrderHistory(orderHistory)
}

const getOrderHistory = () => {
    try {
        return JSON.parse(window.localStorage.orderHistory ?? '[]')
    } catch {
        console.error('Failed to parse orderHistory')
        return []
    }
}

const getOrderIndexOfOrderHistory = (orderId, orderHistory = null) => {
    const orderHistoryInternal = orderHistory || getOrderHistory()
    return orderHistoryInternal.findIndex(e => e.id === orderId)
}

export {
    mapToOrderHistoryDto,
    saveOrderHistory,
    saveCreatedOrderInStorage,
    getOrderHistory,
    getOrderIndexOfOrderHistory,
}
