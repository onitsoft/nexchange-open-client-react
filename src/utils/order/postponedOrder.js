const removePostponedOrder = () => {
    sessionStorage.removeItem('postponedOrderData')
}

const popPostponedOrder = () => {
    const postponedOrderData = JSON.parse(sessionStorage.getItem('postponedOrderData'))
    sessionStorage.removeItem('postponedOrderData')
    return postponedOrderData
}

const savePostponedOrder = orderData => {
    sessionStorage.setItem('postponedOrderData', JSON.stringify(orderData))
}

const getPostponedOrder = () => JSON.parse(sessionStorage.getItem('postponedOrderData'))

const checkIfHasPostponedOrder = () => Boolean(sessionStorage.getItem('postponedOrderData'))

export {
    savePostponedOrder,
    checkIfHasPostponedOrder,
    getPostponedOrder,
    popPostponedOrder,
    removePostponedOrder,
}
