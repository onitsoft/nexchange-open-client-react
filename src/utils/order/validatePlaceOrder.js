import { getPostponedOrder, removePostponedOrder, savePostponedOrder } from './postponedOrder';
import { isUserResidenceLocationLegal } from '../restrictedCountires/restirctionCheck';
import { isUserAuthenticated, isResidenceLocationMissing } from '../user/authChecks';

const PlaceOrderFailureReason = {
  NOT_AUTH: 'NOT_AUTH', // -> to sign up
  MISSING_RESIDENCE_LOCATION: 'MISSING_RESIDENCE_LOCATION', // -> set residence location
  RESTRICTED_RESIDENCE_LOCATION: 'RESTRICTED_RESIDENCE_LOCATION', // -> show error modal
};

const handleAuthorizationToPlaceOrder = async ({ loadUserDetails }) => {
  if (!isUserAuthenticated()) {
    return {
      isUserAuthorizedToPlaceOrder: false,
      placeOrderFailureReason: PlaceOrderFailureReason.NOT_AUTH,
    };
  }

  const userData = await loadUserDetails();

  if (isResidenceLocationMissing(userData.username)) {
    return {
      isUserAuthorizedToPlaceOrder: false,
      placeOrderFailureReason: PlaceOrderFailureReason.MISSING_RESIDENCE_LOCATION,
      userData,
    };
  }

  if (!isUserResidenceLocationLegal(userData.residenceLocation)) {
    removePostponedOrder();
    return {
      isUserAuthorizedToPlaceOrder: false,
      placeOrderFailureReason: PlaceOrderFailureReason.RESTRICTED_RESIDENCE_LOCATION,
      userData,
    };
  }

  return { isUserAuthorizedToPlaceOrder: true, userData };
};

const handlePlaceOrderValidation = async ({ loadUserDetails, submittedOrderData }) => {
  savePostponedOrder(submittedOrderData);
  const { isUserAuthorizedToPlaceOrder, placeOrderFailureReason, userData } = await handleAuthorizationToPlaceOrder({ loadUserDetails });

  return {
    isUserAuthorizedToPlaceOrder,
    placeOrderFailureReason,
    userData
  };
};

const resumeToAuthOrder = async (history, loadUserDetails, redirectRouteOnError = '/') => {
  const postponedOrderData = getPostponedOrder();

  if (!postponedOrderData) {
    console.error('There is no order to resume in.');
    history.push(redirectRouteOnError);
    return;
  }

  await handlePlaceOrderValidation({ submittedOrderData: postponedOrderData, loadUserDetails} );
};

export { PlaceOrderFailureReason, resumeToAuthOrder, handlePlaceOrderValidation };
