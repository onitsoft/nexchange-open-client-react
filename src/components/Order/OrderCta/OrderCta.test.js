import React from 'react';
import { shallow } from 'enzyme';
import OrderCta from './OrderCta';
import order from 'Mocks/order';
import configureStore from 'redux-mock-store';

describe('OrderCta', () => {
    const mockStore = configureStore();
    let store, wrapShallow;
    order.email = 'test@test.com'
    console.log(order)
    beforeEach(() => {
        store = mockStore(order);
        
        wrapShallow = shallow(<OrderCta store={store} order={order} />)
          .dive()
          .dive();
      });
    
      it('renders correctly', () => {
        expect(wrapShallow).toMatchSnapshot();
      });

    
});
