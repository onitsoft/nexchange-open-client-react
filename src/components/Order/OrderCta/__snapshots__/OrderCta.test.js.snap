// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderCta renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
  >
    [Function]
  </I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        false,
        undefined,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          false,
          undefined,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "storeSubscription": undefined,
    },
    Symbol(enzyme.__providerValues__): Map {},
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
